worker/runtime/backend.go

```diff
@@ -140,6 +164,33 @@ func (b *GardenBackend) Ping() (err error) {
 func (b *GardenBackend) Create(gdnSpec garden.ContainerSpec) (garden.Container, error) {
 	ctx := context.Background()
 
+	cont, err := b.createContainer(ctx, gdnSpec)
+	if err != nil {
+		return nil, fmt.Errorf("new container: %w", err)
+	}
+
+	err = b.startTask(ctx, cont)
+	if err != nil {
+		return nil, fmt.Errorf("starting task: %w", err)
+	}
+
+	return NewContainer(
+		cont,
+		b.killer,
+		b.rootfsManager,
+	), nil
+}
+
+func (b *GardenBackend) createContainer(ctx context.Context, gdnSpec garden.ContainerSpec) (containerd.Container, error) {
+	acquired, err := b.checkContainerCapacity(ctx)
```

Comment:
@jamieklassen and I were discussing this. We think separating the responsibility of acquiring the lock from checking the container limit might be more readable. 

In this case we would always acquire the lock and defer the release. Then separately check the container limit (whether we're below the limit OR if there's no limit).